Bundling raw typescript fails

`Worker.create` does not support bundling ESM written in TypeScript (imports of `.ts` files with `.js`):

**workflows.ts**
```ts
import * from './users/workflows.js'; // This file does not exist
```

**users/workflows.ts**
```ts
export async function workflow() {
  // ...
}
```

```
Module not found: Error: Can't resolve './users/workflows.js' in '<app>'

resolve './users/workflows.js' in '<app>/src'
  using description file: <app>/package.json (relative path:

    Field 'browser' doesn't contain a valid alias configuration
    using description file: <app>/package.json (relative pat

      no extension
        Field 'browser' doesn't contain a valid alias configuration
        <app>/src/users/workflows.js doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        <app>/src/users/workflows.js.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        <app>/src/users/workflows.js.js doesn't exist
      as directory
        <app>/src/users/workflows.js doesn't exist
 @ ./src/workflows-autogenerated-entrypoint.cjs 9:9-113

webpack 5.88.2 compiled with 1 error in 1005 ms
```

Current workaround:

```ts
const worker = await Worker.create({
  workflowsPath: require.resolve("./workflows.js"),
  bundlerOptions: {
    webpackConfigHook(config) {
      config.resolve.extensionAlias = { ".js": [".ts", ".js"] }; // Workaround
      return config;
    },
  },
  taskQueue: "app",
  activities,
});
```

Source https://www.npmjs.com/package/resolve-typescript-plugin

--- a/packages/worker/src/workflow/bundler.ts
+++ b/packages/worker/src/workflow/bundler.ts
@@ -201,6 +201,7 @@ exports.importInterceptors = function importInterceptors() {
         // https://webpack.js.org/configuration/resolve/#resolvemodules
         modules: [path.resolve(__dirname, 'module-overrides'), 'node_modules'],
         extensions: ['.ts', '.js'],
+        extensionAlias: { '.js': ['.ts', '.js'] },
         alias: {
           __temporal_custom_payload_converter$: this.payloadConverterPath ?? false,
           __temporal_custom_failure_converter$: this.failureConverterPath ?? false,

